libs/wingsdk/src/target-tf-aws/website.ts

```diff
@@ -0,0 +1,125 @@
+import { readdirSync } from "fs";
+import path from "path";
+import { CloudfrontDistribution } from "@cdktf/provider-aws/lib/cloudfront-distribution";
+import { S3Bucket } from "@cdktf/provider-aws/lib/s3-bucket";
+
+import { S3BucketWebsiteConfiguration } from "@cdktf/provider-aws/lib/s3-bucket-website-configuration";
+import { S3Object } from "@cdktf/provider-aws/lib/s3-object";
+import { Construct } from "constructs";
+import mime from "mime-types";
+import { createEncryptedBucket } from "./bucket";
+import { core } from "..";
+import * as cloud from "../cloud";
+import { Json } from "../std";
+
+const INDEX_FILE = "index.html";
+
+/**
+ * AWS implementation of `cloud.Website`.
+ *
+ * @inflight `@winglang/sdk.cloud.IWebsiteClient`
+ */
+export class Website extends cloud.Website {
+  private readonly bucket: S3Bucket;
+
+  constructor(scope: Construct, id: string, props: cloud.WebsiteProps) {
+    super(scope, id, props);
+
+    this.bucket = createEncryptedBucket(this, true, "website-bucket");
+
+    new S3BucketWebsiteConfiguration(this, "bucket-website-configuration", {
+      bucket: this.bucket.bucket,
+      indexDocument: { suffix: INDEX_FILE },
+    });
+
+    this.uploadFiles(this.path);
+
+    // create a cloudFront distribution
+    const distribution = new CloudfrontDistribution(this, "distribution", {
+      enabled: true,
+      ...(this._domain && { aliases: [this._domain] }),
```

Comment:
hmm excuse my typescript confusion its pretty late here :smile: but Im not sure I understand this line. Is it the same as saying 

```js
const additionalCloudFrontProps = this._domain? { aliases: [this._domain] } : {};
const distribution = new CloudfrontDistribution(this, "distribution", {
  enabled: true,
  ...additionalCloudFrontProps
```

Reply:
exactly!